apps/client/src/features/app-settings/panel/sources-panel/import-map/importMapUtils.ts

```diff
@@ -82,3 +83,23 @@ export function getPersistedOptions(): NamedImportMap {
     return namedImportMap;
   }
 }
+
+// add existing custom fields to an importmap if they don't already exist
+export function addExistingCustomFields(importMap: NamedImportMap, customFields: CustomFields): NamedImportMap {
+  const customFieldsInImportMap = importMap.custom;
+
+  // keep all customfieds that are already diffind in the importmap
+  const newCustomFields = new Array(...customFieldsInImportMap);
+  Object.entries(customFields).forEach(([key, { label }]) => {
+    if (
+      !customFieldsInImportMap.some((customFieldFromImportMap) => {
+        return customFieldFromImportMap.ontimeName === key;
+      })
+    ) {
+      // If the custom field dosn't exist in the import map we add it
+      newCustomFields.push({ ontimeName: key, importName: label });
+    }
+  });
+
+  return { ...importMap, custom: newCustomFields };
+}
```

Comment:
I am not entirely confident on this logic here. I could be missing something.
Could be also nice to add a single happy path test just as documentation

`const customFieldsInImportMap = importMap.custom;`

This reassignemnt doesnt seem to do much for us, but could provide the indirection of us thinking that the object is different, I suggest we remove it

`const newCustomFields = new Array(...customFieldsInImportMap);`

I am not sure why do we need to call the Array constructor, is spreading into a new array not enough? `newCustomFields = [...customFieldsInImportMap];`

`return { ...importMap, custom: newCustomFields };`
why do we need to spread in the end? the `importMap` is given by the consumer and used as a reference here.
Would it make more sense for the consumer to get the `importMap` and pass only the `custom` part here?